src/Microsoft.ComponentDetection.Contracts/TypedComponent/PipComponent.cs

```diff
@@ -10,16 +11,26 @@ private PipComponent()
         /* Reserved for deserialization */
     }
 
-    public PipComponent(string name, string version)
+    public PipComponent(string name, string version, string author = null, string license = null)
     {
         this.Name = this.ValidateRequiredInput(name, nameof(this.Name), nameof(ComponentType.Pip));
         this.Version = this.ValidateRequiredInput(version, nameof(this.Version), nameof(ComponentType.Pip));
+        this.Author = author;
+        this.License = license;
     }
 
     public string Name { get; set; }
 
     public string Version { get; set; }
 
+#nullable enable
+    [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
```

Comment:
Because of the way the simplepip resolver handles building the dependency graph, it is not possible to find Author and License info for the dependencies of a package. This tag makes it so these properties aren't serialized when they are null. This avoids having fields in the graph that will always be null.